### PR TITLE
Data science initial implementation

### PR DESCRIPTION
**Pull request by arrubin**

### CODE REVIEW

File: data_science/lib/abtest/analyzer.rb [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+    end
+
+    def confidence_interval_by_cohort
+      cohorts.each_with_object({}) do |cohort, confidence_intervals|
+        confidence_intervals[cohort.name] = cohort.confidence_interval
+      end
+    end
+
+    def chi_squared
+      cohorts.inject(0) do |chi_squared, cohort|
```
Comment (sshao) [inline data_science/lib/abtest/analyzer.rb]: just `cohorts.inject do |...|` should work here

File: data_science/spec/abtest/analyzer_spec.rb [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+        analyzer = Analyzer.new(test_data)
+        expect(analyzer.chi_squared).to eq(1.82)
+      end
+    end
+
+    describe "#significant?" do
+      it "returns false when difference is not significant" do
+        analyzer = Analyzer.new(test_data)
+        expect(analyzer.significant?).to be false
+      end
```
Comment (sshao) [inline data_science/spec/abtest/analyzer_spec.rb]: might want to check that it returns true when difference *is* significant, too

File: data_science/lib/abtest/analyzer.rb [hunk 3]
```diff
@@ -0,0 +1,81 @@
...
+      cohorts.each_with_object({}) do |cohort, standard_errors|
+        standard_errors[cohort.name] = cohort.standard_error
+      end
+    end
+
+    def confidence_interval_by_cohort
+      cohorts.each_with_object({}) do |cohort, confidence_intervals|
+        confidence_intervals[cohort.name] = cohort.confidence_interval
+      end
+    end
```
Comment (sshao) [inline data_science/lib/abtest/analyzer.rb]: do you need all of these `*_by_cohort` methods, or are you just providing them as helpers to the user? in the case of the latter, it just seems like it'd be easier to provide `cohorts` as an attr_reader or something, and let the user manipulate `cohorts` themself to get these values.

Comment (arrubin) [inline data_science/lib/abtest/analyzer.rb]: Originally most of the logic for the calculations was in these methods. After I extracted the Cohort class they make less sense. I am reconsidering what the API for the analyzer should be.

**Comment by sshao** [comment]
This is solid. Consider my couple of comments, especially the analyzer's public API, but you don't need to resubmit.

**Pull request closed without merging**